openapi/db/dbmodel.py

```diff
@@ -62,17 +67,59 @@ def get_query(self, table, query, consumer=None, params=None):
                 if not isinstance(result, (list, tuple)):
                     result = (result,)
                 filters.extend(result)
+        return filters
+
+    def filter_query(self, table, query, consumer=None, params=None):
+        filters = self.get_filter_list(
+            table,
+            consumer=consumer,
+            params=params,
+        )
         if filters:
             filters = and_(*filters) if len(filters) > 1 else filters[0]
-            query = query.where(filters)
+            return query.where(filters)
+        return query
 
+    def limit_and_order_query(
+            self,
+            table,
+            query,
+            limit=DEF_PAGINATION_LIMIT,
+            offset=0,
+            order_by=None,
+            order_desc=False,
+    ):
         if isinstance(query, Select):
-            # ordering
             query = self.get_order_clause(table, query, order_by, order_desc)
-
-            # pagination
             query = query.offset(offset)
             query = query.limit(limit)
+        return query
+
+    def get_filtered_query(self, table, query, consumer=None, params=None):
+        params, limit_params = self.get_params(params)
+        query = self.filter_query(
+            table,
+            query,
+            consumer=consumer,
+            params=params,
+        )
+        return query
+
+    def get_query(self, table, query, consumer=None, params=None):
+        params, limit_params = self.get_params(params)
+
+        query = self.filter_query(
+            table,
+            query,
+            consumer=consumer,
+            params=params,
+        )
+
+        query = self.limit_and_order_query(
+            table,
+            query,
+            **limit_params,
+        )
```

Comment:
Too many new methods.
`get_query`, `get_filtered_query`, `filter_query`, etc...
It is confusing.